core/reactor_threaded.c

```diff
@@ -1014,18 +1014,227 @@ int worker_thread_count = 0;
 bool _lf_logical_tag_completed = false;
 
 /**
- * Worker thread for the thread pool.
- * This acquires the mutex lock and releases it to wait for time to
- * elapse or for asynchronous events and also releases it to execute reactions.
+ * Return true if the worker should stop now; false otherwise.
```

Comment:
Need to document whether this function assumes that the mutex is held. I think it does, in which case, we have by convention been adding the suffix `_locked` to the function name.